Mathlib/Algebra/Category/ModuleCat/ChangeOfRings.lean

```diff
@@ -852,6 +877,107 @@ instance preservesColimitRestrictScalars {R S : Type*} [Ring R] [Ring S]
+  simp
+
+lemma extendScalarsId_hom_app_one_tmul (M : ModuleCat R) (m : M) :
+    (extendScalarsId R).hom.app M ((1 : R) ⊗ₜ m) = m := by
+  rw [← extendRestrictScalarsAdj_homEquiv_apply,
+    homEquiv_extendScalarsId]
+  dsimp
+
+section
+
+variable {R₁ R₂ R₃ R₄ : Type u₁} [CommRing R₁] [CommRing R₂] [CommRing R₃] [CommRing R₄]
+  (f₁₂ : R₁ →+* R₂) (f₂₃ : R₂ →+* R₃) (f₃₄ : R₃ →+* R₄)
+
+/-- The extension of scalars by a composition of commutative ring morphisms
+identify to the composition of the extension of scalars functors. -/
+noncomputable def extendScalarsComp :
+    extendScalars (f₂₃.comp f₁₂) ≅ extendScalars f₁₂ ⋙ extendScalars f₂₃ :=
+  (conjugateIsoEquiv
+    ((extendRestrictScalarsAdj f₁₂).comp (extendRestrictScalarsAdj f₂₃))
+    (extendRestrictScalarsAdj (f₂₃.comp f₁₂))).symm (restrictScalarsComp f₁₂ f₂₃).symm
+
+lemma homEquiv_extendScalarsComp (M : ModuleCat R₁) :
+    (extendRestrictScalarsAdj (f₂₃.comp f₁₂)).homEquiv _ _
+      ((extendScalarsComp f₁₂ f₂₃).hom.app M) =
+      (extendRestrictScalarsAdj f₁₂).unit.app M ≫
+        (restrictScalars f₁₂).map ((extendRestrictScalarsAdj f₂₃).unit.app _) ≫
+        (restrictScalarsComp f₁₂ f₂₃).inv.app _ := by
+  dsimp [extendScalarsComp, conjugateIsoEquiv, conjugateEquiv]
+  simp only [Category.assoc, Category.id_comp, Category.comp_id,
+    Adjunction.comp_unit_app, Adjunction.homEquiv_unit,
+    Functor.map_comp, Adjunction.unit_naturality_assoc,
+    Adjunction.right_triangle_components]
+  rfl
+
+lemma extendScalarsComp_hom_app_one_tmul (M : ModuleCat R₁) (m : M) :
+    (extendScalarsComp f₁₂ f₂₃).hom.app M ((1 : R₃) ⊗ₜ m) =
+      (1 : R₃) ⊗ₜ[R₂,f₂₃] ((1 : R₂) ⊗ₜ[R₁,f₁₂] m) := by
+  rw [← extendRestrictScalarsAdj_homEquiv_apply, homEquiv_extendScalarsComp]
+  rfl
+
+@[reassoc]
+lemma extendScalars_assoc :
+    (extendScalarsComp (f₂₃.comp f₁₂) f₃₄).hom ≫ whiskerRight (extendScalarsComp f₁₂ f₂₃).hom _ =
+      (extendScalarsComp f₁₂ (f₃₄.comp f₂₃)).hom ≫ whiskerLeft _ (extendScalarsComp f₂₃ f₃₄).hom ≫
+        (Functor.associator _ _ _).inv := by
+  ext M m
+  dsimp
+  have := extendScalarsComp_hom_app_one_tmul (f₂₃.comp f₁₂) f₃₄ M m
+  erw [this, extendScalarsComp_hom_app_one_tmul f₁₂ (f₃₄.comp f₂₃) M m,
+    extendScalarsComp_hom_app_one_tmul f₂₃ f₃₄, ExtendScalars.map_tmul,
+    extendScalarsComp_hom_app_one_tmul f₁₂ f₂₃]
+
+/-- The associativity compatibility for the extension of scalars, in the exact form
+that is needed in the definition `CommRingCat.moduleCatExtendScalarsPseudofunctor`
+in the file `Algebra.Category.ModuleCat.Pseudofunctor` -/
+lemma extendScalars_assoc' :
+    (extendScalarsComp (f₂₃.comp f₁₂) f₃₄).hom ≫ whiskerRight (extendScalarsComp f₁₂ f₂₃).hom _ ≫
+      (Functor.associator _ _ _).hom ≫ whiskerLeft _ (extendScalarsComp f₂₃ f₃₄).inv ≫
+        (extendScalarsComp f₁₂ (f₃₄.comp f₂₃)).inv = 𝟙 _ := by
+  rw [extendScalars_assoc_assoc]
+  simp only [Iso.inv_hom_id_assoc, ← whiskerLeft_comp_assoc, Iso.hom_inv_id,
+    whiskerLeft_id', Category.id_comp]
```

Comment:
I didn't think of this before, but you having to write this lemma makes me think that `map₂_associator` for `Pseudofunctor` should not be written in that form, but rather like `extendScalars_assoc`. But that is besides this PR.

Reply:
If you don't, I might make a follow-up PR to refactor that at some point, but I am quite busy nowadays for a bit.